### ISSUE TITLE
Why systemd Restart=on-success rule ?

### ISSUE THREAD

**Issue by sberthier** [atosatto/ansible-minio#35]
I wonder why the restart rule on systemd service is set to 'on-success'.
I would expect a policy like 'always'. 

I am starting to use minio cluster and during tests, some nodes crashed, either because of out of memory, or with 'Write failed. Insufficient number of disks online' errors when I shutdown some nodes (2 over 4, which is enough to read, but not to write)
Not sure if the last is expected, but I would expect the cluster to be up even read only.

The comment above this rule in minio.service.j2 does not explain why.
```
# Let systemd restart this service only if it has ended with the clean exit code or signal.
Restart=on-success
```

**Comment by atosatto** [comment]
Hi @sberthier, this is a fair point.

Originally this was meant to avoid systemd to restart the service in case of crash.
However I do also see that the systemd unit definition available at https://github.com/minio/minio-service/blob/master/linux-systemd/minio.service is setting `Restart=always`.

It would probably make sense to align our systemd unit template to the one maintained upstream.

@SuperQ thoughts?

**Comment by SuperQ** [comment]
Yea, "always" seems reasonable to me. We should make sure there's a sleep/backoff setting in there to avoid noisy crashloops.

**Comment by atosatto** [comment]
Closed with #36
